lib/vauth/gsasl.c

```diff
@@ -38,81 +39,75 @@
+    return false;
   }
 
-  res = gsasl_client_start(gsasl->ctx, "SCRAM-SHA-1", &gsasl->client);
+  res = gsasl_client_start(gsasl->ctx, mech, &gsasl->client);
   if(res != GSASL_OK) {
-    infof(data, "SCRAM start (%d):%s\n", res, gsasl_strerror(res));
-    return CURLE_REMOTE_ACCESS_DENIED;
+    gsasl_done(gsasl->ctx);
+    return false;
   }
 
-  gsasl_property_set(gsasl->client, GSASL_AUTHID, userp);
-  gsasl_property_set(gsasl->client, GSASL_PASSWORD, passwdp);
-
-  res = gsasl_step64(gsasl->client, NULL, outptr);
-  if(res != GSASL_NEEDS_MORE) {
-    infof(data, "SCRAM cf step (%d):%s\n", res, gsasl_strerror(res));
-    return CURLE_REMOTE_ACCESS_DENIED;
-  }
-  *outlen = strlen(*outptr);
-
-  return CURLE_OK;
+  return true;
 }
 
-CURLcode Curl_auth_create_scram_cl_message(struct Curl_easy *data,
-                                           const char *chlg64,
-                                           struct gsasldata *gsasl,
-                                           char **outptr, size_t *outlen)
+CURLcode Curl_auth_gsasl_start(struct Curl_easy *data,
+                               const char *userp,
+                               const char *passwdp,
+                               struct gsasldata *gsasl)
 {
-  int res;
-
-  res = gsasl_step64(gsasl->client, chlg64, outptr);
-  if(res != GSASL_NEEDS_MORE) {
-    infof(data, "SCRAM cl step (%d):%s\n", res, gsasl_strerror(res));
-    return CURLE_REMOTE_ACCESS_DENIED;
-  }
-
-  *outlen = strlen(*outptr);
+  gsasl_property_set(gsasl->client, GSASL_AUTHID, userp);
+  gsasl_property_set(gsasl->client, GSASL_PASSWORD, passwdp);
 
   return CURLE_OK;
 }
 
-
-CURLcode Curl_auth_verify_scram_sl_message(struct Curl_easy *data,
-                                           const char *chlg64,
-                                           struct gsasldata *gsasl,
-                                           char **outptr, size_t *outlen)
+CURLcode Curl_auth_gsasl_token(struct Curl_easy *data,
+                               const char *chlg64,
+                               struct gsasldata *gsasl,
+                               char **outptr, size_t *outlen)
 {
-  int res;
+  unsigned char *chlg = NULL;
+  size_t chlglen = 0;
+  size_t chlg64len = chlg64 ? strlen(chlg64) : 0;
+  int result;
+  char *response;
+
+  if(chlg64){
+    result = Curl_base64_decode(chlg64, &chlg, &chlglen);
+    if(result)
+      return result;
+  }
 
-  res = gsasl_step64(gsasl->client, chlg64, outptr);
-  if(res != GSASL_OK) {
-    infof(data, "SCRAM sl step (%d):%s\n", res, gsasl_strerror(res));
-    return CURLE_REMOTE_ACCESS_DENIED;
+  result = gsasl_step(gsasl->client, chlg, chlglen, &response, outlen);
+  if(result != GSASL_OK && result != GSASL_NEEDS_MORE){
+    if (chlg64)
+      free (chlg);
+    infof(data, "GSASL step: %s\n", gsasl_strerror(result));
+    return CURLE_BAD_CONTENT_ENCODING;
   }
 
-  *outlen = strlen(*outptr);
+  if (*outlen > 0)
+    {
```

Comment:
Not our indent style!

Reply:
Now 'make checksrc' is happy.  Is there any tool to just re-indent everything to your style, rather than fixing things manually?

Reply:
Unfortunately we don't. I once tried to make a clang-format config that matches our style but I only just *almost* succeeded: https://github.com/curl/curl/wiki/clang-format-style